static/src/javascripts/projects/common/modules/identity/ad-prefs.js

```diff
@@ -0,0 +1,123 @@
+// @flow
+
+import React, { Component } from 'react';
+import { render } from 'react-dom';
+import fastdom from 'lib/fastdom-promise';
+import {
+    getAdConsentState,
+    setAdConsentState,
+    allAdConsents,
+} from './ad-prefs.lib';
+import type { AdConsent } from './ad-prefs.lib';
+
+const rootSelector: string = '.js-manage-account__ad-prefs';
+
+type ConsentRadioButtonProps = {
+    value: string,
+    label: string,
+    checked: boolean,
+    consent: AdConsent,
+    onToggle: () => void,
+};
+
+class ConsentRadioButton extends Component<ConsentRadioButtonProps, {}> {
+    handleChange(event): void {
```

Comment:
Could you add a type to `event`?

Reply:
phew, this was a bit more involved than I thought it would be! `SyntheticInputEvent<HTMLInputElement>`